examples/test_config.c

```diff
@@ -0,0 +1,169 @@
+/* test_config.c
+ *
+ * Copyright (C) 2006-2021 wolfSSL Inc.
+ *
+ * This file is part of wolf Key Manager.
+ *
+ * wolfKeyMgr is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfKeyMgr is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include "examples/test_config.h"
+
+static EtsiClientCtx* gEtsiClient = NULL;
+
+/* ETSI Client Helpers */
+int test_esti_client_connect(const char* urlStr)
+{
+    int ret = 0;
+    static char urlStrCopy[HTTP_MAX_URI];
+    static HttpUrl url;
+
+    /* setup key manager connection */
+    if (gEtsiClient == NULL) {
+        wolfEtsiClientInit();
+
+        gEtsiClient = wolfEtsiClientNew();
+        if (gEtsiClient) {
+            wolfEtsiClientAddCA(gEtsiClient, ETSI_TEST_CLIENT_CA);
+            wolfEtsiClientSetKey(gEtsiClient,
+                ETSI_TEST_CLIENT_KEY, ETSI_TEST_CLIENT_PASS,
+                ETSI_TEST_CLIENT_CERT, WOLFSSL_FILETYPE_PEM);
+
+            if (urlStr) {
+                strncpy(urlStrCopy, urlStr, HTTP_MAX_URI);
+                memset(&url, 0, sizeof(url));
+                wolfHttpUrlDecode(&url, urlStrCopy);
+            }
+
+            ret = wolfEtsiClientConnect(gEtsiClient, url.domain, url.port,
+                ETSI_TEST_TIMEOUT_MS);
+            if (ret != 0) {
+                printf("Error connecting to ETSI server %s! %d\n", urlStr, ret);
+                test_etsi_client_cleanup();
+            }
+        }
+        else {
+            ret = WOLFKM_BAD_MEMORY;
+        }
+    }
+    return ret;
+}
+
+int test_etsi_client_get(const char* urlStr, EtsiKey* key, int keyType)
+{
+    int ret;
+    const char* keyStr = wolfEtsiKeyGetTypeStr(keyType);
+    if (keyStr == NULL) {
+        return WOLFKM_NOT_COMPILED_IN;
+    }
+
+    ret = test_esti_client_connect(urlStr);
+    if (ret == 0 && key != NULL) {
+        /* Get and set a static ephemeral for each supported key type */
+        ret = wolfEtsiClientGet(gEtsiClient, key, keyType, NULL, NULL,
+            ETSI_TEST_TIMEOUT_MS);
+        /* positive return means new key returned */
```

Comment:
These three lines are redundant as they are already in the comments below.  Maybe this belongs where the function is written.